ding/reward_model/curiosity_reward_model.py

```diff
@@ -0,0 +1,176 @@
+from typing import Union, Tuple
+from easydict import EasyDict
+
+import random
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import torch.nn.functional as F
+
+from ding.utils import SequenceType, REWARD_MODEL_REGISTRY
+from ding.model import FCEncoder, ConvEncoder
+from .base_reward_model import BaseRewardModel
+
+def collect_states(iterator: list) -> Tuple[list, list, list] : 
+    states = []
+    next_states = []
+    actions = []
+    for item in iterator:
+        state = item['obs']
+        next_state = item['next_obs']
+        action = item['action']
+        states.append(state)
+        next_states.append(next_state)
+        actions.append(action)
+    return states, next_states, actions 
+
+# Intrinsic Curiosity Model (ICM Module)
+class ICMNetwork(nn.Module):
+
+    def __init__(self, obs_shape: Union[int, SequenceType], hidden_size_list: SequenceType, action_shape:int) -> None:
+        super(ICMNetwork, self).__init__()
+        if isinstance(obs_shape, int) or len(obs_shape) == 1:
+            self.feature= FCEncoder(obs_shape, hidden_size_list)
+        elif len(obs_shape) == 3:
+            self.feature= ConvEncoder(obs_shape, hidden_size_list)
+        else:
+            raise KeyError(
+                "not support obs_shape for pre-defined encoder: {}, please customize your own ICM model".
+                format(obs_shape)
+            )
+        self.action_shape = action_shape
+        feature_output = hidden_size_list[-1]
+        self.inverse_net = nn.Sequential(
+            nn.Linear(feature_output * 2, 512),
+            nn.ReLU(),
+            nn.Linear(512, action_shape)
+        )
+        self.residual = [nn.Sequential(
+            nn.Linear(action_shape + 512, 512),
+            nn.LeakyReLU(),
+            nn.Linear(512, 512),
+        ) for _ in range(8)]
+        self.forward_net_1 = nn.Sequential(
+            nn.Linear(action_shape + feature_output, 512),
+            nn.LeakyReLU()
+        )
+        self.forward_net_2 = nn.Sequential(
+            nn.Linear(action_shape + 512, feature_output),
+        )
+
+    def forward(self, state: torch.Tensor, next_state: torch.Tensor, action_long: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+        state = torch.FloatTensor(state)
+        next_state = torch.FloatTensor(next_state)
+        action_long = torch.LongTensor(action_long)
+        action_onehot = torch.FloatTensor(len(action_long), self.action_shape)
+        action_onehot.zero_()
+        action_onehot.scatter_(1, action_long.view(len(action_long), -1), 1)
+        action = action_onehot
+        encode_state = self.feature(state)
+        encode_next_state = self.feature(next_state)
+        # get pred action
+        pred_action = torch.cat((encode_state, encode_next_state), 1)
+        pred_action = self.inverse_net(pred_action)
```

Comment:
This variable should be named as `logit` action, it is not `action` but `logit`:
- `action = torch.multinomial(torch.softmax(logit, dim=1), 1)`